### ISSUE TITLE
Suggestion: SCP client with the key

### ISSUE THREAD

**Issue by Sysa** [ronf/asyncssh#166]
Hello,

I didn't find how to pass key to `asyncssh.scp`, even with `read_private_key`, and looking into source code shows me that it is not supported as `client_keys` in `ssh`

I understand that most of cases of `scp` can be covered by `sftp` and `ssh` combination, but still : is it possible to use `scp` in `asyncssh` with the key or not?

Thanks,
Alex

**Comment by ronf** [comment]
There are a couple of ways to do this in the released version of AsyncSSH:

1. Put the client keys you want to use in ~/.ssh/authorized_keys
2. Open a connection to the remote system you want to copy to/from with the connect() or create_connection() call with whatever authentication arguments you need, and then pass that into scp as the source or destination, as a tuple of the form (connection, filename). In cases where you want to use a filename of ".", this can be shortened to just the connection object.

For instance:

```python
    async with asyncssh.connect('remotehost', client_keys=my_keys) as conn:
        await asyncssh.scp('localfile', (conn, 'remote file'))
```

Last weekend I checked in an improvement in the "develop" branch to support all of the optional arguments to connect() in the scp() call, though. So, in cases where you only need one remote connection or where all remote connections can use the same optional connect() arguments, this can be shortened to:

```python
    async with asyncssh.scp('localfile', 'remotehost:remotefile', client_keys=my_keys)
```

This change will be in the next release. See eba209a for more details.

**Comment by Sysa** [comment]
Thank you so much, Ron!

**Comment by ronf** [comment]
This change is now available in AsyncSSH 1.15.0.
